vendor_data.py

```diff
@@ -1,241 +1,375 @@
-        else:
-            with pd.ExcelWriter(filename, mode="a", engine="openpyxl", if_sheet_exists="overlay") as writer:
+        FileName = "_".join([str(self.ProjectName), str(self.SourceLang)]) + ".xlsx" #the name of the excel file, based on arguments `ProjectName` and `SourceLang`
+        df = pd.DataFrame(Data) #make a panda dataframe out of the dictionary
+        if not os.path.exists(FileName): #check if the file exists, if not a new file is written
+            df.to_excel(FileName, index=False, sheet_name="VendorData")
+        else: #if file exists, the data is appended to the existing file
+            with pd.ExcelWriter(FileName, mode="a", engine="openpyxl", if_sheet_exists="overlay") as writer:
                 df.to_excel(writer, sheet_name = "VendorData", startrow=writer.sheets["VendorData"].max_row, header = None, index=False)
+    def ReadExcel(self): #method to read the file with filename {ProjectName_SourceLang} into a dictionary
+        FileName = "_".join([str(self.ProjectName), str(self.SourceLang)]) + ".xlsx"
+        if os.path.exists(FileName):
+            ExcelRecords = pd.read_excel(FileName) #read filename
+            ExcelRecordsDf = ExcelRecords.loc[:, ~ExcelRecords.columns.str.contains('^Unnamed')] #remove any unnamed columns
+            VendorDict = ExcelRecordsDf.to_dict() #turn dataframe into a dictionary
+            return VendorDict #print the Dictionary
+        else:
+            raise ValueError(f"There is no file for {self.ProjectName} in {self.SourceLang}")
+    
+    def ModExcel(self, Key, Index, NewValue):
+        """Method to modify existing vendor in excel file
+
+        Args:
+            Key (str): one of the modifiable keys in VendorData.Keys
+            Index (int): the index of the value that needs to be modified
+            NewValue : the new value of the index, type depends on the key, either str or float
 
-# don't forget to properly dcument them with docstrings!!
+        Raises:
+            ValueError: raised if the key is not one of the keys in VendorData.Keys
+            ValueError: raised if the Index is not valid for a certain key
+        """
+        FileName = "_".join([str(self.ProjectName), str(self.SourceLang)]) + ".xlsx"
+        if os.path.exists(FileName):
+            ExcelRecords = pd.read_excel(FileName)
+            ExcelRecordsDf = ExcelRecords.loc[:, ~ExcelRecords.columns.str.contains('^Unnamed')]
+            VendorDict = ExcelRecordsDf.to_dict()
+            if not Key in self.Keys:
+                raise ValueError("Invalid key, run 'VendorData.Keys' to see options.")
+            if not Index in VendorDict[Key]:
+                raise ValueError("Invalid index, run 'self.ReadExcel()' to see options")
+            if Key == "E-mail": #validate e-mail address if a value for the key E-mail is modified
+                CheckVendorMail(NewValue)
+            if Key == "Word Rate": #validate word rate if a value for the key Word Rate is modified

```

Comment:
For all the `Key ==...` checks, I would chain them with `elif`. Otherwise, it will check if `Key == "Word Rate"` even if it already say that it is "E-mail"